RyuSocks.Generator/Packet/AccessorGenerator.Array.cs

```diff
@@ -0,0 +1,192 @@
+                source.AppendLine($"this[{packetField.GetOffset()} + i] = {maybeCastValue}value[i];");
+                source.LeaveScope();
+            }
+
+            return source.GetLines();
+        }
+
+        private static string[] GenerateArraySByteAccessor(PacketFieldModel packetField, bool isGetter)
+        {
+            BlockBuilder source = new();
+
+            if (isGetter)
+            {
+                string fieldTypeName = packetField.FieldType.Name.Extract(0, -2);
+
+                source.AppendLine($"{fieldTypeName}[] result = new {fieldTypeName}[{packetField.GetLength()}];");
+                source.AppendLine();
+                source.EnterScope($"for (int i = 0; i < {packetField.GetLength()}; i++)");
+                source.AppendLine($"result[i] = ({fieldTypeName})this[{packetField.GetOffset()} + i];");
+                source.LeaveScope();
+                source.AppendLine();
+                source.AppendLine("return result;");
+            }
+            else
+            {
+                source.EnterScope($"for (int i = 0; i < {packetField.GetLength()}; i++)");
+                source.AppendLine($"this[{packetField.GetOffset()} + i] = (byte)value[i];");
+                source.LeaveScope();
+            }
+
+            return source.GetLines();
+        }
+
+        private static string[] GenerateArrayIntegralAccessor(PacketFieldModel packetField, TypeConverterModel converter, bool isGetter)
+        {
+            BlockBuilder source = new();
+
+            if (isGetter)
+            {
+                string fieldTypeName = packetField.FieldType.Name.Extract(0, -2);
+                string maybeCast = packetField.FieldType.IsEnum ? $"({fieldTypeName})" : string.Empty;
+
+                source.AppendLine($"{fieldTypeName}[] result = new {fieldTypeName}[{packetField.GetLength()}];");
+                source.AppendLine();
+                source.EnterScope($"for (int i = 0; i < {packetField.GetLength()}; i++)");
+                source.AppendLine($"result[i] = {maybeCast}{converter.GetReaderName(packetField.IsBigEndian)}(this.AsSpan({packetField.GetOffset()} + (i * {converter.Length}), {converter.Length}));");
+                source.LeaveScope();
+                source.AppendLine();
+                source.AppendLine("return result;");
+            }
+            else
+            {
+                string valueParameter = packetField.FieldType.IsEnum ? $"({packetField.FieldType.ActualType.ToTypeString()})value" : "value";
+
+                source.EnterScope($"for (int i = 0; i < {packetField.GetLength()}; i++)");
+                source.AppendLine($"{converter.GetWriterName(packetField.IsBigEndian)}(this.AsSpan({packetField.GetOffset()} + (i * {converter.Length}), {converter.Length}), {valueParameter}[i]);");
+                source.LeaveScope();
+            }
+
+            return source.GetLines();
+        }
```

Comment:
In all of these 3 lines the Block Builder gets set up the same way around the one main changing line, is it possible to make this a function?

Reply:
I extracted all the common code to `GenerateSimpleArrayAccessor()` and simplified the other methods instead.